### PR TITLE
Master note newapi rba

### PR DESCRIPTION
**Pull request by rba-odoo**
Task: https://www.odoo.com/web#id=13294&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.ZG9OnTSXxZbnLSYM

### CODE REVIEW

File: addons/note/models/note.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                if stage.user_id == self.env.user:
+                    record.stage_id = stage
+
+    name = fields.Text(compute='_get_note_first_line', string='Note Summary', store=True)
+    user_id = fields.Many2one('res.users', string='Owner', default=lambda self: self.env.uid)
+    memo = fields.Html(string='Note Content')
+    sequence = fields.Integer()
+    stage_id = fields.Many2one('note.stage', compute='_get_stage_per_user', inverse='_set_stage_per_user', default=_get_default_stage_id, string='Stage')
+    stage_ids = fields.Many2many('note.stage', 'note_stage_rel', 'note_id', 'stage_id', string='Stages of Users')
+    open = fields.Boolean(string='Active', track_visibility='onchange', default=1)
```
Comment (hmo-odoo) [inline addons/note/models/note.py]: do not use python keywords. so replace with is_open

File: addons/note/models/note.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models
+from openerp.tools import html2plaintext
+
+class NoteStage(models.Model):
```
Comment (hmo-odoo) [inline addons/note/models/note.py]: can you make new python file for note.stage model ?

File: addons/note/models/note.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    _name = "note.stage"
+    _description = "Note Stage"
+    _order = 'sequence asc'
+
+    name = fields.Char(string='Stage Name', translate=True, required=True)
+    sequence = fields.Integer(help="Used to order the note stages", default=1)
+    user_id = fields.Many2one('res.users', string='Owner', help="Owner of the note stage.", required=True, ondelete='cascade', default=lambda self: self.env.uid)
+    fold = fields.Boolean(string='Folded by Default', default=0)
+
+class NoteTag(models.Model):
```
Comment (hmo-odoo) [inline addons/note/models/note.py]: can you make new python file for note.tag ?

File: addons/note/models/note.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+    def _set_stage_per_user(self):
+        for record in self:
+            if record.stage_id:
+                record.stage_ids = record.stage_id + record.stage_ids.filtered(lambda stage: stage.user_id != self.env.user)
+
+    @api.depends('stage_ids.user_id')
+    def _get_stage_per_user(self):
+        for record in self:
+            for stage in record.stage_ids:
```
Comment (hmo-odoo) [inline addons/note/models/note.py]: should use filtered

File: addons/note/models/res_config.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import fields, models
+
+#upgrade config setting page to configure pad
+class BaseConfiguration(models.TransientModel):
+    """upgrade config setting page to configure pad"""
+    _inherit = 'base.config.settings'
```
Comment (hmo-odoo) [inline addons/note/models/res_config.py]: can you change name of python file: base_config_setting.py ?

File: addons/note/models/res_users.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, models
+
+class ResUsers(models.Model):
+    _name = 'res.users'
```
Comment (hmo-odoo) [inline addons/note/models/res_users.py]: remove _name

File: addons/note/models/note.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+class NoteTag(models.Model):
+    _name = "note.tag"
+    _description = "Note Tag"
+
+    name = fields.Char(string='Tag Name', required=True)
+
+class NoteNote(models.Model):
+    """ Note """
+    _name = 'note.note'
```
Comment (hmo-odoo) [inline addons/note/models/note.py]: python file should be note_note.py

File: addons/note/models/res_users.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    _inherit = ['res.users']
+
+    @api.model
+    def create(self, data):
+        user = super(ResUsers, self).create(data)
+        stage_ids = self.env['note.stage'].sudo().search([('user_id', '=', self.env.uid)])
```
Comment (hmo-odoo) [inline addons/note/models/res_users.py]: do not change logic during migrate. keep current behavior.

Comment (rba-odoo) [inline addons/note/models/res_users.py]: @hmo-odoo Done.!

File: addons/note/models/note.py (hunk 7 above)
Comment (rba-odoo) [inline addons/note/models/note.py]: @hmo-odoo Done.!

File: addons/note/models/res_users.py (hunk 6 above)
Comment (rba-odoo) [inline addons/note/models/res_users.py]: @hmo-odoo Done.!

File: addons/note/models/res_config.py (hunk 5 above)
Comment (rba-odoo) [inline addons/note/models/res_config.py]: @hmo-odoo Done.!

File: addons/note/models/note.py (hunk 4 above)
Comment (rba-odoo) [inline addons/note/models/note.py]: @hmo-odoo Done.!

File: addons/note/models/note.py (hunk 3 above)
Comment (rba-odoo) [inline addons/note/models/note.py]: @hmo-odoo Done.!

File: addons/note/models/note.py (hunk 2 above)
Comment (rba-odoo) [inline addons/note/models/note.py]: @hmo-odoo Done.!

File: addons/note/data/note_data.xml [hunk 9]
```diff
@@ -1,36 +1,34 @@
-<?xml version="1.0"?>
-<openerp>
-  <data>
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
 
-    <record model="note.stage" id="note_stage_00">
+    <record id="note_stage_00" model="note.stage">
```
Comment (tde-banana-odoo) [inline addons/note/data/note_data.xml]: Let's have a convention when renaming data: note_stage_data_01. Demo will be: note_stage_demo_01.

File: addons/note/models/base_config_setting.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import fields, models
+
+#upgrade config setting page to configure pad
```
Comment (tde-banana-odoo) [inline addons/note/models/base_config_setting.py]: Remove this unnecessary comment.

File: addons/note/models/base_config_setting.py [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+class BaseConfiguration(models.TransientModel):
+    """upgrade config setting page to configure pad"""
```
Comment (tde-banana-odoo) [inline addons/note/models/base_config_setting.py]: Remove this unnecessary comment.

File: addons/note/models/note_note.py [hunk 12]
```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models
+from openerp.tools import html2plaintext
+
+class NoteNote(models.Model):
```
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: Some remarks 
1/ put compute fields after the field declaration
2/ put them in the fields declaration order

File: addons/note/models/note_note.py [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+
+    #read the first line (convert html into text)
+    @api.one
+    @api.depends('memo')
+    def _get_note_first_line(self):
+        self.name = (self.memo and html2plaintext(self.memo) or "").strip().replace('*', '').split("\n")[0]
+
+    # return the default stage for the uid user
+    def _get_default_stage_id(self):
+        return self.env['note.stage'].search([('user_id', '=', self.env.uid)], limit=1).id or False
```
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: Is the "or False" really necessary ?

File: addons/note/models/note_note.py [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+
+    def _set_stage_per_user(self):
+        for record in self:
+            if record.stage_id:
+                record.stage_ids = record.stage_id + record.stage_ids.filtered(lambda stage: stage.user_id != self.env.user)
```
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: 1/ I don't understand this code, and it seems weird
2/ Use |
3/ What the hell is the purpose of this code ?

File: addons/note/models/note_note.py (hunk 12 above)
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: 3/ compute method should be _compute_<field_name>, inverse _set_<field_name>

File: addons/note/models/note_note.py [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+
+    @api.depends('stage_ids.user_id')
+    def _get_stage_per_user(self):
+        for record in self:
+            for stage in record.stage_ids.filtered(lambda stage: stage.user_id == self.env.user):
+                record.stage_id = stage
+
+    name = fields.Text(compute='_get_note_first_line', string='Note Summary', store=True)
```
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: Shouldn't the name be a char field ?

File: addons/note/models/note_note.py [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+    user_id = fields.Many2one('res.users', string='Owner', default=lambda self: self.env.uid)
+    memo = fields.Html(string='Note Content')
+    sequence = fields.Integer()
+    stage_id = fields.Many2one('note.stage', compute='_get_stage_per_user', inverse='_set_stage_per_user', default=_get_default_stage_id, string='Stage')
+    stage_ids = fields.Many2many('note.stage', 'note_stage_rel', 'note_id', 'stage_id', string='Stages of Users')
+    is_open = fields.Boolean(string='Active', track_visibility='onchange', default=1)
```
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: Why is this field not named active, like all active fields ?

File: addons/note/models/note_note.py [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+    date_done = fields.Date()
```
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: Please add a string.

File: addons/note/models/note_note.py (hunk 12 above)
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: 4/ action method should be action_<my_action>

File: addons/note/models/note_note.py [hunk 18]
```diff
@@ -0,0 +1,112 @@
...
+    color = fields.Integer('Color Index')
+    tag_ids = fields.Many2many('note.tag', 'note_tags_rel', 'note_id', 'tag_id', string='Tags')
+
+    @api.one
+    def onclick_note_is_done(self):
```
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: 1/ Batch
2/ action_is_done

File: addons/note/models/note_note.py [hunk 19]
```diff
@@ -0,0 +1,112 @@
...
+        self.is_open = 0
+        self.date_done = fields.Date.today()
+
+    @api.one
+    def onclick_note_not_done(self):
```
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: 1/ Batch
2/ action_undone
3/ reset date_done

File: addons/note/models/note_note.py [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+        self.is_open = 1
+
+    #writing method (no modification of values)
+    @api.model
+    def name_create(self, name):
```
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: What does it add compared to a create with a default value on stage_ids ?

File: addons/note/models/note_note.py [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+        record = self.create({'memo': name, 'stage_ids': [(6, 0, [self.env.context['default_stage_id']])]})
+        return record.name_get()[0]
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, context=None, orderby=False, lazy=True):
```
Comment (tde-banana-odoo) [inline addons/note/models/note_note.py]: TL;DR

File: addons/note/models/res_users.py [hunk 22]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, models
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    @api.model
+    def create(self, data):
```
Comment (tde-banana-odoo) [inline addons/note/models/res_users.py]: Crip crap crapy, crapy crapy crap ...

Remove this thing.

File: addons/note/views/note_note_views.xml [hunk 23]
```diff
@@ -1,8 +1,7 @@
-<?xml version="1.0"?>
-<openerp>
-  <data>
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (tde-banana-odoo) [inline addons/note/views/note_note_views.xml]: While you are at it, rename the xml_ids following the convention.

File: addons/note/data/note_data.xml (hunk 9 above)
Comment (rba-odoo) [inline addons/note/data/note_data.xml]: @tde-banana-odoo  Done.!

File: addons/note/models/base_config_setting.py (hunk 10 above)
Comment (rba-odoo) [inline addons/note/models/base_config_setting.py]: @tde-banana-odoo  Done.!

File: addons/note/models/base_config_setting.py (hunk 11 above)
Comment (rba-odoo) [inline addons/note/models/base_config_setting.py]: @tde-banana-odoo  Done.!

File: addons/note/models/note_note.py (hunk 12 above)
Comment (rba-odoo) [inline addons/note/models/note_note.py]: @tde-banana-odoo  Done.!

File: addons/note/models/note_note.py (hunk 13 above)
Comment (rba-odoo) [inline addons/note/models/note_note.py]: @tde-banana-odoo  Done.!

File: addons/note/models/note_note.py (hunk 14 above)
Comment (rba-odoo) [inline addons/note/models/note_note.py]: @tde-banana-odoo  Not required anymore after refactoring stage mechanism

File: addons/note/models/note_note.py (hunk 15 above)
Comment (rba-odoo) [inline addons/note/models/note_note.py]: @tde-banana-odoo  Done.!

File: addons/note/models/note_note.py (hunk 18 above)
Comment (rba-odoo) [inline addons/note/models/note_note.py]: @tde-banana-odoo  Done.!

File: addons/note/models/note_note.py (hunk 19 above)
Comment (rba-odoo) [inline addons/note/models/note_note.py]: @tde-banana-odoo  Done.!

File: addons/note/models/note_note.py (hunk 20 above)
Comment (rba-odoo) [inline addons/note/models/note_note.py]: @tde-banana-odoo  Done.!

File: addons/note/models/note_note.py (hunk 21 above)
Comment (rba-odoo) [inline addons/note/models/note_note.py]: @tde-banana-odoo  Done.!

File: addons/note/models/res_users.py (hunk 22 above)
Comment (rba-odoo) [inline addons/note/models/res_users.py]: @tde-banana-odoo  Done.!

File: addons/note/views/note_note_views.xml (hunk 23 above)
Comment (rba-odoo) [inline addons/note/views/note_note_views.xml]: @tde-banana-odoo  Done.!

**Comment by hmo-odoo** [comment]
note module is migrated at https://github.com/odoo/odoo/commit/b3b0b23fe2094812eb9d609b43c04a4a70c60cbf

**Pull request closed without merging**